### PR TITLE
Fix native UserLocation on Android

### PR DESCRIPTION
**Pull request by janicduplessis**
This fixes 2 issues I noticed with native UserLocation on Android

- The LocationComponentManager class is not initialized properly and an exception occurs because mLocationComponent is null. This adds a missing call to the update method to make sure it is initialized before using it.

- The location component appears underneath annotations. This is because the layer id used is out of date with newer mapbox versions. The id is now public in LocationComponentConstants so we can use that directly.

Also do some minor dead code cleanup.

### CODE REVIEW

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java [hunk 1]
```diff
@@ -98,16 +95,8 @@ public boolean hasLocationComponent() {
         return (mLocationComponent != null);
     }
 
-    public void forceLocationUpdate(Location location) {
```
Comment (janicduplessis) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java:101]: Unused

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java [hunk 2]
```diff
@@ -98,16 +95,8 @@ public boolean hasLocationComponent() {
...
-        mLocationComponent.forceLocationUpdate(location);
-    }
-
     public void update(@NonNull Style style) {
-        if (mLocationComponent == null) {
```
Comment (janicduplessis) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java:106]: Same code in both branches

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java [hunk 3]
```diff
@@ -32,8 +30,6 @@
     private LocationComponent mLocationComponent = null;
     private Context mContext = null;
 
-        // state
-    private @CameraMode.Mode int mCameraMode = CameraMode.NONE;
```
Comment (janicduplessis) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/location/LocationComponentManager.java:36]: unused

**Review by mfazekas — APPROVED** [review]

**Comment by ferdicus** [comment]
Thanks for your contribution @janicduplessis

**Pull request merged**